### DISCUSSION TITLE
DLC32 M4 does not work

### DISCUSSION THREAD

**Discussion by Tribo-2dot0** [arkypita/LaserGRBL#1825] [General]
Hello everybody. I bought an MKS DLC32 V2.1 board and at first I managed to configure everything ok, motors move correctly, but I can only make the laser work with M3… if I configure M4, the message “unsupported command” appears… laser is on ($32=1). Does anyone know how I solve this?

**Comment by arkypita** [comment]
Very strange, any idea of what firmare is on it?

Google it, look like a known problem 
![Screenshot_2022-05-14-22-41-46-232_com android chrome](https://user-images.githubusercontent.com/8782035/168447544-4bb18290-5314-42c3-acca-077e82b904c9.jpg)

**Comment by Tribo-2dot0** [comment]
I've looked all over but I haven't found anything... how do I see what firmware I have? Firmware would be the grbl version? I read somewhere that it is 1.1f.

**Comment by Tribo-2dot0** [comment]
Got it... I downloaded a .bin file and the MKS tool and updated it... it seems that everything is ok now.
Just one thing, before the update this appeared:

![20220514_111313](https://user-images.githubusercontent.com/83982525/168449826-079f1db1-ea93-44f6-a6d6-e2dbc621df3d.jpg)

then this appears (similar to how it was in Mana SE):

![20220514_191121](https://user-images.githubusercontent.com/83982525/168449829-005affff-5be0-4ce7-830e-62a03251a19e.jpg)

I don't know which version of 1.1 this is...
Question: can I upgrade to original 1.1h using LaserGrbl?

**Reply by arkypita** [reply]
I think this board use STM32 microprocessor, not possible to flash using grbl. Needs specific procedure as described by the manufacturer

**Reply by Tribo-2dot0** [reply]
Ok, deixarei assim então. Obrigado.
